### DISCUSSION TITLE
REST server not working in docker

### DISCUSSION THREAD

**Discussion by niwim** [walt-id/.github#38] [Help] [answered]
Hi,

I'm trying to follow this guide but I cannot get the REST API to work https://docs.walt.id/v/ssikit/getting-started/rest-apis.

When I run the following command `docker run -itv $(pwd)/data:/app/data -p 7000-7004:7000-7004 waltid/ssikit serve` the container successfully starts but for some reason, I'm not able to access the endpoints 

Running `curl -X 'GET' 'http:/localhost:7001//health' -H 'accept: application/json'` i get error `curl: (56) Recv failure: Connection reset by peer`

The port mapping checks out and I've even tried running `netstat -an|grep LISTEN` which returns the port-mapping as expected.

I'm starting to wonder if there is something wrong with the REST-service in the container build?

**Comment by niwim** [comment]
Hi,

I got it to work by building the docker image via the github repo so suspect something might be off with the Docker Hub image.

**Reply by Python-David** [reply]
Hi, tried this suggested approach and it's still not responding. Did you find a way around this? I'm stuck here too thanks!

**Reply by niwim** [reply, marked as answer]
Hi, 

I got it to work by building the docker image from source. If I remember correctly, the guide doesn't specify the need for opening the ports when executing the docker run command. Have you tried starting the container using the following cmd `docker run -itv $(pwd)/data:/app/data -p 7000-7004:7000-7004 waltid/ssikit serve` instead of the one in the guide.

**Comment by taminobaumann** [comment]
Hey

Another option, when using the docker image from docker hub, would be to add the following flag to the command: `-b 0.0.0.0`

Full run command:
`docker run -p 7000-7004:7000-7004 -itv $(pwd)/data:/app/data waltid/ssikit serve -b 0.0.0.0`
